syncer/syncer_test.go

```diff
@@ -164,16 +163,17 @@ func TestSynchronize_OnlyOneSynchronize(t *testing.T) {
 	var wg sync.WaitGroup
 	wg.Add(1)
 	go func() {
-		require.True(t, ts.syncer.synchronize(context.Background()))
+		require.True(t, ts.syncer.synchronize(ctx))
 		wg.Done()
 	}()
 	<-started
-	require.False(t, ts.syncer.synchronize(context.Background()))
+	require.False(t, ts.syncer.synchronize(ctx))
 	// allow synchronize to finish
 	close(done)
 	wg.Wait()
 
 	require.Equal(t, uint64(1), ts.syncer.run)
+	cancel()
```

Comment:
with the defer this wouldn't be necessary.

Reply:
It is required to cancel the context before calling `Syncer::Close` or it will hang waiting for the errgroup.